### PR TITLE
CommandHandler

### PR DESCRIPTION
**Pull request by MiniDigger**

### CODE REVIEW

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+            }
+            String cmdLabel = buffer.toString();
+
+            // if command exists, execute it
+            if ( commandMap.containsKey( cmdLabel ) ) {
+                Method executorMethod = commandMap.get( cmdLabel ).getKey();
+                Object executorObject = commandMap.get( cmdLabel ).getValue();
+                CommandInfo commandInfo = executorMethod.getAnnotation( CommandInfo.class );
+                if ( !commandInfo.perm().equals( "" ) && !commandSender.hasPermission( commandInfo.perm() ) ) {
+                    //TODO Send no permission message
```
Comment (MiniDigger) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java:142]: TODO

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+                    return true;
+                }
+                if ( commandInfo.allowConsole() && !( commandSender instanceof Player ) ) {
+                    //TODO Send no console message
```
Comment (MiniDigger) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java]: TODO

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    public void registerHelp() {
+        Set<HelpTopic> help = new TreeSet<>( HelpTopicComparator.helpTopicComparatorInstance() );
+        for ( String commandLabel : commandMap.keySet() ) {
+            if ( !commandLabel.contains( "." ) ) {
+                org.bukkit.command.Command cmd = bukkitCommandMap.getCommand( commandLabel );
+                HelpTopic topic = new GenericCommandHelpTopic( cmd );
+                help.add( topic );
+            }
+        }
+        //TODO which perm node is requiered to be able to see the help? Where do we get that perm node from? Hardcode?
```
Comment (MiniDigger) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java]: TODO

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java (hunk 2 above)
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java]: No Console message should be hardcoded to english

File: Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java (hunk 3 above)
Comment (geNAZt) [inline Bukkit-Commons/src/main/java/net/devintia/commons/bukkit/command/CommandHandler.java]: I would suggest (system).command.(commandname).help. For example raid.command.start.help for "/raid start" help

**Pull request merged**